frontendFile/src/main/java/com/virtuslab/gitmachete/frontend/file/MacheteCompletionContributor.java

```diff
@@ -0,0 +1,63 @@
+package com.virtuslab.gitmachete.frontend.file;
+
+import com.intellij.codeInsight.completion.CompletionContributor;
+import com.intellij.codeInsight.completion.CompletionParameters;
+import com.intellij.codeInsight.completion.CompletionResultSet;
+import com.intellij.codeInsight.completion.PlainPrefixMatcher;
+import com.intellij.codeInsight.lookup.LookupElementBuilder;
+import com.intellij.openapi.progress.ProgressManager;
+import com.intellij.psi.PsiFile;
+import git4idea.repo.GitRepositoryManager;
+import io.vavr.collection.List;
+
+public class MacheteCompletionContributor extends CompletionContributor {
+
+  @Override
+  public void fillCompletionVariants(CompletionParameters parameters, CompletionResultSet result) {
+    PsiFile file = parameters.getOriginalFile();
+    var project = file.getProject();
+
+    var gitRepository = List.ofAll(GitRepositoryManager.getInstance(project).getRepositories())
+        .find(r -> file.getVirtualFile().getPath().startsWith(r.getRoot().getPath()));
+
+    if (gitRepository.isEmpty()) {
+      return;
+    }
+
+    var branchNames = List.ofAll(gitRepository.get().getInfo().getLocalBranchesWithHashes().keySet())
+        .map(localBranch -> localBranch.getName());
+
+    /**
+     * {@link CompletionResultSet#stopHere} marks the result set as stopped.
+     * Completion service calls contributors as long as everyone gets called or result set get marked as stopped.
+     * The following call allows to avoid other contributor invocations (performance).
+     *
+     * See {@link com.intellij.codeInsight.completion.CompletionService#getVariantsFromContributors}
+     */
+    result.stopHere();
+
+    String prefix = getCompletionPrefix(parameters);
+    var matcher = new PlainPrefixMatcher(prefix, /* prefixMatchesOnly */ true);
+    var completionResultSet = result.caseInsensitive().withPrefixMatcher(matcher);
+    for (String branchName : branchNames) {
+      ProgressManager.checkCanceled();
+      completionResultSet.addElement(LookupElementBuilder.create(branchName));
+    }
+  }
+
+  public static String getCompletionPrefix(CompletionParameters parameters) {
+    String text = parameters.getOriginalFile().getText();
+    int offset = parameters.getOffset();
+    return getCompletionPrefix(text, offset);
+  }
+
+  private static String getCompletionPrefix(String text, int offset) {
+    int lastSpaceIdx = text.lastIndexOf(' ', offset - 1) + 1;
+    int lastTabIdx = text.lastIndexOf('\t', offset - 1) + 1;
```

Comment:
...and pls drop a comment why the original `getCompletionPrefix` wasn't viable here